vector-im/element-android
# FTUE - Registration Captcha and T&Cs screens

## Type of change


## Content

Fixes #5279

- Extracts the captcha webview logic to a reusable class 
- Creates new `Captcha` and `Terms` fragments with the updated FTUE styling and copies  _legacy style_ variants of the originals to use when the new FTUE feature flag is disabled
- Updates the `PolicyItem` for both new and legact views to match the updated compound designs

## Motivation and context

To have a consistent style across the registration steps

## Screenshots / GIFs

| GIF | 
| --- | 
![after-captcha-terms](https://user-images.githubusercontent.com/1848238/162749504-e1f8a231-4ad3-4b9d-9f46-dca2ff0c6038.gif)

| LIGHT CAPTCHA | DARK CAPTCHA | LIGHT TERMS | DARK TERMS
| --- | --- | --- | --- | 
![Screenshot_20220411_095633](https://user-images.githubusercontent.com/1848238/162740135-dae16aa7-b549-4ac5-91a8-d20263be9849.png)|![Screenshot_20220411_095649](https://user-images.githubusercontent.com/1848238/162740133-edfcfca8-7f69-4aeb-9533-84af1941e090.png)|![Screenshot_20220411_095715](https://user-images.githubusercontent.com/1848238/162740130-5371a414-becf-4451-8948-ac6e79319e33.png)|![Screenshot_20220411_095703](https://user-images.githubusercontent.com/1848238/162740132-b847af2b-ccf2-4267-88eb-38603ecec253.png)


| SMALL CAPTCHA | SMALL TERMS | 
| --- | --- |
![Screenshot_20220411_102639](https://user-images.githubusercontent.com/1848238/162740129-a423fa73-b288-4e68-9933-3222da39501b.png)|![Screenshot_20220411_102722](https://user-images.githubusercontent.com/1848238/162740126-77c8195e-1bf8-424f-8856-f97c4dad33d4.png)|


| TABLET PORTRAIT | TABLET LANDSCAPE |
| --- | --- |   
![Screenshot_20220411_112937](https://user-images.githubusercontent.com/1848238/162740053-75ab5c00-c094-41f9-aee4-73562e0482d9.png)|![Screenshot_20220411_112944](https://user-images.githubusercontent.com/1848238/162740052-637d2061-fba2-4f1a-9563-3d7858eeb7bb.png)
![Screenshot_20220411_113008](https://user-images.githubusercontent.com/1848238/162740043-442b909f-a966-4d09-aab3-ca28c0171488.png)|![Screenshot_20220411_113002](https://user-images.githubusercontent.com/1848238/162740047-bac434fe-df46-48de-a3f9-2b665577b6d2.png)


| BEFORE LEGACY TERMS | AFTER LEGACY TERMS |
| --- | --- |
![before-legacy-terms](https://user-images.githubusercontent.com/1848238/162751198-90cf778d-dadb-42a0-a9d8-f779ed58d433.png)|![after-legacy-terms](https://user-images.githubusercontent.com/1848238/162751202-7d5361e5-bd29-4c70-bbdf-64f4c76c9bb7.png)


## Tests

- Enable the combined registration feature flag
- Create a new account
- Notice the captcha and terms screens have been updated

## Tested devices

- OS version(s): 29, 31
